README.md

```diff
@@ -43,13 +43,34 @@ npm test
 
 ## Building from source
 
-Compiled from https://github.com/facebook/zstd/tree/dev/contrib/single_file_libs, with the
+Compiled from https://github.com/facebook/zstd/tree/dev/build/single_file_libs, with the
 following steps:
 
 ```shell
 ./combine.sh -r ../../lib -o zstddeclib.c zstddeclib-in.c
-emcc zstddeclib.c -Oz -s EXPORTED_FUNCTIONS="['_ZSTD_decompress', '_ZSTD_findDecompressedSize', '_ZSTD_isError', '_malloc', '_free']" -s ALLOW_MEMORY_GROWTH=1 -s MALLOC=emmalloc -o zstddec.wasm
-base64 zstddec.wasm > zstddec.txt
+```
+
+Add the following to the end of *zstddeclib.c* to [shim out unused wasi
+imports](https://github.com/emscripten-core/emscripten/issues/17331),
+
+```c
+int __wasi_fd_write(int, int, int, int) {
+  __builtin_trap();
+}
+int __wasi_fd_seek(int, int64_t, int, int) {
+  __builtin_trap();
+}
+int __wasi_fd_close(int) {
+  __builtin_trap();
+}
+_Noreturn void  __wasi_proc_exit(int) {
+  __builtin_trap();
+}
+```
+
+```shell
+emcc zstddeclib.c -Oz -s EXPORTED_FUNCTIONS="['_ZSTD_decompress', '_ZSTD_findDecompressedSize', '_ZSTD_isError', '_malloc', '_free']"  -Wl,--no-entry -s ALLOW_MEMORY_GROWTH=1 -s MALLOC=emmalloc -o zstddec.wasm
+base64 -w 0 zstddec.wasm > zstddec.txt
```

Comment:
Unfortunately, the size of the build is roughly doubled after the update. Any idea what might be the cause of that?

Reply:
Hmm, not sure why that is case.

Reply:
I'm afraid this may be a blocker -- I don't know if I can justify updating the dependency with that size difference. Many users of the package are sensitive to bundle size.

Reply:
I pushed improved flags and also backed down on the version. Here is what I found:

```
The size of the resulting zstddec.wasm increases in subsequent versions:

1.5.0: 32k
1.5.1: 40k
1.5.5: 52k
```